src/plugins/plugin-http.js

```diff
@@ -0,0 +1,118 @@
+'use strict';
+var shimmer = require('shimmer');
+var url = require('url');
+var isString = require('is').string;
+var merge = require('lodash.merge');
+var httpAgent = require('_http_agent');
+
+function patchModuleRoot (request, api) {
+  var labels = api.labels;
+  function wrapInvocation (options, callback) {
+    return function httpActionTrace () {
+      if (!this._google_trace_patched && options) {
+        // Don't keep wrapping our same request
+        this._google_trace_patched = true;
+        return patchHTTPRequest(options, callback, request);
+      }
+      return request.apply(this, arguments);
+    }
+  }
+
+  function setTraceHeader (parsedOptions, context) {
+    if (context) {
+      return merge(parsedOptions, {
+        headers: {
+          [TraceLabels.TRACE_CONTEXT_HEADER_NAME]: context
+        }
+      });
+    }
+    return parsedOptions;
+  }
+
+  function parseRequestOptions (requestOptions) {
+    return isString(requestOptions) ?
+      merge(url.parse(requestOptions), {headers: {}}) :
+      merge({headers: {}}, requestOptions);
+  }
+
+  function extractUrl (parsedOptions) {
+    var uri = parsedOptions;
+    var agent = parsedOptions._defaultAgent || httpAgent.globalAgent;
+    return isString(uri) ? uri :
+      (parsedOptions.protocol || agent.protocol) + '//' +
+      (parsedOptions.hostname || parsedOptions.host || 'localhost') +
+      ((isString(parsedOptions.port) ? (':' + parsedOptions.port) : '')) +
+      (parsedOptions.path || parseRequestOptions.pathName || '/');
+  }
+
+  function getSpanName (requestOptions) {
+    if (isString(options)) {
+      options = url.parse(requestOptions);
+    }
+    // c.f. _http_client.js ClientRequest constructor
+    return options.hostname || options.host || 'localhost';
+  }
+
+  function patchHTTPRequest (requestOptions, callback, request) {
+    var parsedOptions = parseRequestOptions(requestOptions);
+    var uri = extractUrl(parsedOptions);
+    var transaction = api.createTransaction({
```

Comment:
Since this is an outgoing request, we do not want to create a new transaction. In general we don't want the same code path to create both a root and child span.